src/pages/workspace/rules/RulesCustomNamePage.tsx

```diff
@@ -43,17 +46,52 @@ function RulesCustomNamePage({route}: RulesCustomNamePageProps) {
 
     const customNameDefaultValue = Str.htmlDecode(policy?.fieldList?.[CONST.POLICY.FIELDS.FIELD_LIST_TITLE].defaultValue ?? '');
 
-    const validateCustomName = ({customName}: FormOnyxValues<typeof ONYXKEYS.FORMS.RULES_CUSTOM_NAME_MODAL_FORM>) => {
-        const errors: FormInputErrors<typeof ONYXKEYS.FORMS.RULES_CUSTOM_NAME_MODAL_FORM> = {};
-        if (!customName) {
-            errors[INPUT_IDS.CUSTOM_NAME] = translate('common.error.fieldRequired');
-        } else if (customName.length > CONST.WORKSPACE_NAME_CHARACTER_LIMIT) {
-            errors[INPUT_IDS.CUSTOM_NAME] = translate('common.error.characterLimitExceedCounter', {
-                length: customName.length,
-                limit: CONST.WORKSPACE_NAME_CHARACTER_LIMIT,
-            });
+    const validateCustomName = useCallback(
+        ({customName}: FormOnyxValues<typeof ONYXKEYS.FORMS.RULES_CUSTOM_NAME_MODAL_FORM>) => {
+            const errors: FormInputErrors<typeof ONYXKEYS.FORMS.RULES_CUSTOM_NAME_MODAL_FORM> = {};
+            if (!customName) {
+                errors[INPUT_IDS.CUSTOM_NAME] = translate('common.error.fieldRequired');
+            } else if (customName.length > CONST.WORKSPACE_NAME_CHARACTER_LIMIT) {
+                errors[INPUT_IDS.CUSTOM_NAME] = translate('common.error.characterLimitExceedCounter', {
+                    length: customName.length,
+                    limit: CONST.WORKSPACE_NAME_CHARACTER_LIMIT,
+                });
+            }
+            return errors;
+        },
+        [translate],
+    );
+
+    const clearFieldListError = () => {
+        clearPolicyErrorField(policyID, 'fieldList');
+    };
+
+    // Get the actual error from the nested structure
+    const fieldListErrors = policy?.errorFields?.fieldList as Record<string, unknown> | undefined;
+    const textTitleErrors = fieldListErrors?.[CONST.POLICY.FIELDS.FIELD_LIST_TITLE] as Record<string, unknown> | undefined;
+    const defaultValueErrors = textTitleErrors?.defaultValue as Record<string, string> | undefined;
```

Comment:
Can we improve the type definitions of this so that we don't have to cast everything?